### PR TITLE
Add nifm service

### PR DESCRIPTION
**Pull request by roblabla**
Allows getting the IP address of the switch.

### CODE REVIEW

File: lib/ipc/nifm.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+static ipc_object_t nifm_object;
+
+ipc_object_t nifm_g_object;
+
+result_t nifm_init() {
+	result_t r;
+	ipc_request_t rq = ipc_default_request;
+	ipc_response_fmt_t rs = ipc_default_response_fmt;
+
+
```
Comment (misson20000) [inline lib/ipc/nifm.c]: whitespace

File: lib/ipc/nifm.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	ipc_object_t nifm_s_object;
+	r = sm_get_service(&nifm_s_object, "nifm:s");
+	if(r)
+		return r;
+
+	rq.request_id = 4;
+	rs.num_move_handles = 1;
+	rs.move_handles = &(nifm_g_object.session);
+	nifm_g_object.object_id = -1;
```
Comment (misson20000) [inline lib/ipc/nifm.c]: use `rs.num_objects` and `rs.objects` (they will still work for non-domains and will properly initialize your `ipc_object_t`)

File: lib/ipc/nifm.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	return 0;
+}
+
+result_t nifm_get_ip_address(u32 *ip) {
+	ipc_request_t rq = ipc_default_request;
+	ipc_response_fmt_t rs = ipc_default_response_fmt;
+
+	rq.request_id = 12;
+	rs.raw_data = ip;
+	rs.raw_data_size = sizeof(u32);
```
Comment (misson20000) [inline lib/ipc/nifm.c]: `sizeof(*ip)` makes the meaning a little more clear

File: lib/ipc/nifm.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#include<libtransistor/types.h>
+#include<libtransistor/ipc.h>
+#include<libtransistor/ipc/nifm.h>
+#include<libtransistor/ipc/sm.h>
+
+static ipc_object_t nifm_object;
+
+ipc_object_t nifm_g_object;
```
Comment (misson20000) [inline lib/ipc/nifm.c]: make this `static` (nothing outside this file has any business with it)

File: lib/ipc/nifm.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
+#include<libtransistor/types.h>
+#include<libtransistor/ipc.h>
+#include<libtransistor/ipc/nifm.h>
+#include<libtransistor/ipc/sm.h>
+
+static ipc_object_t nifm_object;
```
Comment (misson20000) [inline lib/ipc/nifm.c]: is this unused?

**Comment by roblabla** [comment]
Fixed

**Pull request merged**